[Caffe2] Fix merge logic bug

scramsby opened the pull request:
Summary: `ExprGroup::getMergeCandidates()` had a logic bug. The vector being initialized had its arguments mis-ordered. This didn't trigger a build warning because the warning about implicit cast from an integral type to `bool` wasn't enabled.

Test Plan: `buck test fbsource//arvr/mode/win/vs2019/cuda11/opt fbsource//arvr/mode/hybrid_execution //arvr/libraries/neural_net_inference/TorchScript/...`

Differential Revision: D41488939

davidberard98 commented:
cc @jjsjann123

scramsby commented:
@pytorchbot rebase -s

jjsjann123 reviewed (approved):
Ah.. Thanks for patching this!

scramsby committed:
[Caffe2] Fix merge logic bug (#89551)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/89551

`ExprGroup::getMergeCandidates()` had a logic bug. The vector being initialized had its arguments mis-ordered. This didn't trigger a build warning because the warning about implicit cast from an integral type to `bool` wasn't enabled.

Test Plan: `buck test fbsource//arvr/mode/win/vs2019/cuda11/opt fbsource//arvr/mode/hybrid_execution //arvr/libraries/neural_net_inference/TorchScript/...`

Reviewed By: davidberard98

Differential Revision: D41488939

fbshipit-source-id: e3b1c4a80109845c402a9d02506669f1618884b3

davidberard98 commented:
@pytorchbot rebase -s
